### PR TITLE
Refactored vrm search. Introduced potential for reuse

### PR DESCRIPTION
**Pull request by przemos**

### CODE REVIEW

File: webapp/src/main/java/uk/gov/dvsa/motr/remote/vehicledetails/VehicleDetails.java [hunk 1]
```diff
@@ -7,22 +7,29 @@
...
     @JsonProperty("regNumber")
     private String regNumber;
+
     @JsonProperty("yearOfManufacture")
     private Integer yearOfManufacture;
+
     @JsonDeserialize(using = LocalDateDeserialiser.class)
-    @JsonProperty("motExpiryDate")
+    @JsonProperty(value = "motExpiryDate")
     private LocalDate motExpiryDate;
```
Comment (Thomas-Franklin) [inline webapp/src/main/java/uk/gov/dvsa/motr/remote/vehicledetails/VehicleDetails.java]: Is there a reason to specify value= on this property, but not the rest of them? Just more a question for understanding when and when not to use. :)

File: webapp/src/test/java/uk/gov/dvsa/motr/web/resource/EmailResourceTest.java [hunk 2]
```diff
@@ -28,11 +28,11 @@ public void emailTemplateIsRenderedOnGet() throws Exception {
     @Test
     public void onPostWithValidEmailPageReloaded() throws Exception {
 
-        // Temporary test until DB integration and redirect is in place
+        // Temporary throwsClientExceptionWhenInvalidUrl until DB integration and redirect is in place
         TemplateEngineStub engine = new TemplateEngineStub();
         EmailResource resource = new EmailResource(engine);
 
-        assertEquals(RESPONSE, resource.emailPagePost("test@test.com"));
+        assertEquals(RESPONSE, resource.emailPagePost("throwsClientExceptionWhenInvalidUrl@throwsClientExceptionWhenInvalidUrl.com"));
```
Comment (przemos) [inline webapp/src/test/java/uk/gov/dvsa/motr/web/resource/EmailResourceTest.java]: fix names

File: webapp/src/test/java/uk/gov/dvsa/motr/web/test/aws/TestLambdaContext.java [hunk 3]
```diff
@@ -9,9 +9,13 @@
 
 public class TestLambdaContext implements Context {
 
+    public static final Context INSTANCE = new TestLambdaContext();
+
+    public static final String AWS_REQUEST_ID = UUID.randomUUID().toString();
```
Comment (przemos) [inline webapp/src/test/java/uk/gov/dvsa/motr/web/test/aws/TestLambdaContext.java]: remove it as not relevant

File: webapp/src/test/java/uk/gov/dvsa/motr/web/validator/EmailValidatorTest.java [hunk 4]
```diff
@@ -30,7 +30,7 @@
     @DataProvider
     public static Object[][] validEmails() {
         return new Object[][]{
-                {"test@test.com"},
+                {"throwsClientExceptionWhenInvalidUrl@throwsClientExceptionWhenInvalidUrl.com"},
```
Comment (przemos) [inline webapp/src/test/java/uk/gov/dvsa/motr/web/validator/EmailValidatorTest.java]: naming fix

File: webapp/src/test/java/uk/gov/dvsa/motr/web/viewmodel/ReviewViewModelTest.java [hunk 5]
```diff
@@ -67,7 +67,7 @@ public void whenExpiryDateIsSetCorrectFormatIsReturned() {
     @Test
     public void whenEmailIsSetItCanBeRetrieved() {
 
-        String expected = "test@test.com";
+        String expected = "throwsClientExceptionWhenInvalidUrl@throwsClientExceptionWhenInvalidUrl.com";
```
Comment (przemos) [inline webapp/src/test/java/uk/gov/dvsa/motr/web/viewmodel/ReviewViewModelTest.java]: naming fix

**Pull request merged**